### PR TITLE
Add image injection feature to new Appium structure

### PR DESCRIPTION
**Pull request by eyaly**

### CODE REVIEW

**Comment by eyaly** [comment]
@nadvolod - As you asked, please take a look . For now I added only image injection . 
Users can run this example by the command line: 
mvn clean install -Dregion=eu -DtestngXmlFile=image_injection_ios_test.xml
They can change:
reign to "us" to run on US DC
testngXmlFile - the testNG xml file
feature - the feature to run (the default, for now, is image_injection )
rdc - for running on RDC or EmuSim ( image injection run only on RDC so there is no use for that) 

In this way, users can run only one feature from the Appium examples on any OS they want

File: appium-testng-examples/src/test/java/helpers/Utils.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            base64Image = Base64.getEncoder().encodeToString(imageData);
+        } catch (FileNotFoundException e) {
+            System.out.println("Image not found" + e);
+        } catch (IOException ioe) {
+            System.out.println("Exception while reading the Image " + ioe);
+        }
+        return base64Image;
+    }
+
+    public static boolean isAndroidBrowserOpened(AndroidDriver driver) throws InterruptedException {
```
Comment (nadvolod) [inline appium-testng-examples/src/test/java/helpers/Utils.java:34]: @eyaly these static methods won't work correctly when ran in parallel

File: appium-testng-examples/pom.xml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    <dependencies>
+
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+            <version>${testng.version}</version>
+        </dependency>
+
+        <!-- Selenium -->
+        <dependency>
```
Comment (nadvolod) [inline appium-testng-examples/pom.xml:38]: @eyaly don't need Selenium dependency as those are located in our repository [level POM.xml](https://github.com/saucelabs-training/demo-java/blob/d730b3a7bb565793b671cdbd80d4ecc40ea4502e/pom.xml#L54). The nice thing about this setup is that we just reuse those dependencies :) Same goes for Appium and maven-compiler-plugin

**Review by nadvolod — COMMENTED** [review]
@eyaly Yes! Much cleaner and less code to maintain. It's almost ready, just a few points in comments below. Also,
1. Do we have old example of this elsewhere in the repo that needs to be removed? I seem to recall that in https://github.com/saucelabs-training/demo-java/pull/71
2. Need to resolve the Codacy issues, let me know if you're not able to see each one.
3. Can we get a quick README on how to run these tests please?

File: appium-testng-examples/pom.xml (hunk 2 above)
Comment (eyaly) [reply appium-testng-examples/pom.xml]: Thanks @nadvolod for the feedback. I finished with all the changes and the fix according to your suggestions. Please review and merge

**Comment by nadvolod** [comment]
@eyaly looking great, looks like there are 2 remaining issues. Codacy issues and there is a merge conflict in pom.xml

**Comment by eyaly** [comment]
@nadvolod - About Codacy. The errors are not something that I should fix
It is about defining the elements and the error is about: Use explicit scoping instead of the default package private level

**Comment by eyaly** [comment]
And about the pom.xml conflict @nadvolod . 
If I resolve the conflicts, I will marge the PR (that's what happen to you last time) . 
In the modules - this is my new one:
        <module>selenium-testng-examples</module>
And I removed:
        <module>java8/selenium3/examples/testng/image-injection</module>
        <module>java8/selenium3/examples/testng/biometric-login</module>

This is the list of modules that should be:
    <modules>
        <module>best-practice</module>
        <module>common</module>
        <module>on-boarding-modules/junit</module>
        <module>on-boarding-modules/testng</module>
        <module>parallel-testing/cross-browser-and-platform-config</module>
        <module>parallel-testing/single-browser-config</module>
        <module>headless</module>
        <module>appium-junit4-examples</module>
        <module>w3c-examples/w3c-junit5</module>
        <module>w3c-examples/w3c-testng</module>
        <module>java11/selenium3-junit5-examples</module>
        <module>java8/selenium3.junit4.examples.cucumber</module>
        <module>appium-junit4-cucumber-examples</module>
        <module>java8/visual-e2e/visuale2e.junit4.examples</module>
        <module>java11/selenium3.junit4.examples</module>
        <module>java8/selenium.junit4.examples</module>
        <module>java8/selenium.junit3.examples</module>
        <module>java8/selenium4.junit4.examples</module>
        <module>best-practice</module>
        <module>appium-testng-examples</module>
        <module>selenium-testng-best-practice</module>
        <module>selenium-testng-examples</module>
    </modules>

**Pull request merged**